test/e2e/alertmanager_test.go

```diff
@@ -1839,17 +1839,33 @@ func testAMRollbackManualChanges(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	replicas := int32(0)
-	sset.Spec.Replicas = &replicas
-	if _, err := ssetClient.Update(context.Background(), sset, metav1.UpdateOptions{}); err != nil {
+	sset.Spec.Replicas = func(i int32) *int32 { return &i }(0)
```

Comment:
nit: can't we use "k8s.io/utils/pointer" here?
```suggestion
	sset.Spec.Replicas = pointer.Int32(0)
```